### ISSUE TITLE
typescript - api-cors-lambda-crud-dynamodb: unable to hit endpoints

### ISSUE THREAD

**Issue by leopic** [aws-samples/aws-cdk-examples#261]
### The Question
After successfully running `cdk deploy` on this example, hitting the provided end points always returns an error:
```bash
 ✅  ApiLambdaCrudDynamoDBExample

Outputs:
ApiLambdaCrudDynamoDBExample.itemsApiEndpoint8392E274 = https://n8pjnrg6h4.execute-api.us-east-1.amazonaws.com/prod/

Stack ARN:
arn:aws:cloudformation:us-east-1:REDACTED_ACCOUNT_ID:stack/ApiLambdaCrudDynamoDBExample/ee04e9b0-6de1-11ea-86aa-0aece6d977c1
```

A `GET` request fails with the following message:
```json
{
message: "Missing Authentication Token"
}
```

When testing this directly from the Lambda (creating a test event) the method does run correctly.

However, applying the same debugging method to a `POST` request, fails with the following message:
```
Response:
{
  "errorType": "Runtime.ImportModuleError",
  "errorMessage": "Error: Cannot find module 'uuid/v4'",
  "trace": [
    "Runtime.ImportModuleError: Error: Cannot find module 'uuid/v4'",
    "    at _loadUserApp (/var/runtime/UserFunction.js:100:13)",
    "    at Object.module.exports.load (/var/runtime/UserFunction.js:140:17)",
    "    at Object.<anonymous> (/var/runtime/index.js:45:30)",
    "    at Module._compile (internal/modules/cjs/loader.js:778:30)",
    "    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)",
    "    at Module.load (internal/modules/cjs/loader.js:653:32)",
    "    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)",
    "    at Function.Module._load (internal/modules/cjs/loader.js:585:3)",
    "    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)",
    "    at startup (internal/bootstrap/node.js:283:19)"
  ]
}

Request ID:
"f00fbf97-6057-4230-8aa5-4ec442770703"

Function Logs:
START RequestId: f00fbf97-6057-4230-8aa5-4ec442770703 Version: $LATEST
2020-03-24T15:44:01.808Z	undefined	ERROR	Uncaught Exception	{"errorType":"Runtime.ImportModuleError","errorMessage":"Error: Cannot find module 'uuid/v4'","stack":["Runtime.ImportModuleError: Error: Cannot find module 'uuid/v4'","    at _loadUserApp (/var/runtime/UserFunction.js:100:13)","    at Object.module.exports.load (/var/runtime/UserFunction.js:140:17)","    at Object.<anonymous> (/var/runtime/index.js:45:30)","    at Module._compile (internal/modules/cjs/loader.js:778:30)","    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)","    at Module.load (internal/modules/cjs/loader.js:653:32)","    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)","    at Function.Module._load (internal/modules/cjs/loader.js:585:3)","    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)","    at startup (internal/bootstrap/node.js:283:19)"]}
END RequestId: f00fbf97-6057-4230-8aa5-4ec442770703
REPORT RequestId: f00fbf97-6057-4230-8aa5-4ec442770703	Duration: 5138.95 ms	Billed Duration: 5200 ms	Memory Size: 128 MB	Max Memory Used: 29 MB	
Unknown application error occurred
Runtime.ImportModuleError
```

Keep in mind that the timeout for the `POST` request was increased, it used to be 3 seconds, but it was failing every time.

### Environment

  - **CDK CLI Version:** 1.30.0 (build 4f54ff7)
  - **Example:** api-cors-lambda-crud-dynamodb
  - **Example Version:** 1.0.0
  - **OS:** macOS Mojave
  - **Language:** TypeScript

### Other information 
No modifications were made to the source code other than manually increasing the timeout on for the `POST` method handler.

Also, FWIW, when I tried to add my `AccessKey` and `SecretKey` using [Postman](https://docs.aws.amazon.com/apigateway/latest/developerguide/how-to-use-postman-to-call-api.html) I kept getting the following error:
```json
{
    "Message": "User: arn:aws:iam::REDACTEDUSERID:user/REDACTEDUSERNAME is not authorized to access this resource"
}
```

**Comment by codypenta** [comment]
Hey leopic,

This fails because the package (UUID) is not in the default AWS Lambda runtime for Node. 

Lambda spins up a sandbox environment with the bare necessities. These bare necessities are often just the language runtime and its standard library plus AWS essentials like the AWS SDK. 

The two solutions from here would be to use a pseudo-random number generator for ID (not recommended cause the same id could potentially occur) or use [Lambda Layers](https://docs.aws.amazon.com/cdk/api/latest/docs/aws-lambda-readme.html#layers)

When it comes to hitting the API gateway endpoint, make sure to hit the resource endpoint specifically

https://n8pjnrg6h4.execute-api.us-east-1.amazonaws.com/prod/items

**Comment by bminahan73** [comment]
I ran into this as well. The example this issue is referring to just includes the Typescript files under src but not the required node_modules directory or package.json.

**Comment by nazreen** [comment]
@leopic  your second error, I encountered it as well, and I've submitted a PR to address it: #278 

the solution fortunately does not require the `index.ts` code to be changed because `lambda.assetCode()` will also bundle the `node_modules` for deployment

**Comment by NGL321** [comment]
Hey @leopic,

Thank you for bringing this up to our attention, and I apologize for the delay in response. This is the same root cause as the issue @nazreen found. Since he put in work on a fix, I am going to make that issue the tracking issue for this fix.

In short, @codypenta was correct, this was a combination of the `uuid` library implementing some breaking changes, and the library itself not existing in the default lambda runtime.

You can track fix and workarounds on issue #276, I will close this issue for now.

Cheers, and thanks again for reporting!
😸 😷
